chartfx-math/src/test/java/de/gsi/math/spectra/ApodizationTests.java

```diff
@@ -0,0 +1,136 @@
+        assertArrayEquals(BLACKMANNUTTAL8, Apodization.BlackmanNuttall.getWindow(8));
+        assertArrayEquals(EXPONENTIAL8, Apodization.Exponential.getWindow(8));
+        assertArrayEquals(FLATTOP8, Apodization.FlatTop.getWindow(8));
+        assertArrayEquals(HAMMING8, Apodization.Hamming.getWindow(8));
+        assertArrayEquals(HANN8, Apodization.Hann.getWindow(8));
+        assertArrayEquals(HANNEXP8, Apodization.HannExp.getWindow(8));
+        assertArrayEquals(NUTTAL8, Apodization.Nuttall.getWindow(8));
+        // check single values
+        assertEquals(RECTANGULAR8[0], Apodization.Rectangular.getIndex(0, 8));
+        // check single values with additional argument
+        assertEquals(RECTANGULAR8[4], Apodization.Rectangular.getIndex(4, 8, 2.0));
+        assertEquals(BLACKMAN8[4], Apodization.Blackman.getIndex(4, 8, 2.0));
+        assertEquals(BLACKMANHARRIS8[4], Apodization.BlackmanHarris.getIndex(4, 8, 2.0));
+        assertEquals(BLACKMANNUTTAL8[4], Apodization.BlackmanNuttall.getIndex(4, 8, 2.0));
+        assertEquals(EXPONENTIAL8[4], Apodization.Exponential.getIndex(4, 8, 1.0));
+        assertEquals(1.1813604128656459, Apodization.Exponential.getIndex(4, 8, 2.0));
+        assertEquals(FLATTOP8[4], Apodization.FlatTop.getIndex(4, 8, 2.0));
+        assertEquals(HAMMING8[4], Apodization.Hamming.getIndex(4, 8, 2.0));
+        assertEquals(HANN8[4], Apodization.Hann.getIndex(4, 8, 2.0));
+        assertEquals(HANNEXP8[4], Apodization.HannExp.getIndex(4, 8, 2.0));
+        assertEquals(NUTTAL8[4], Apodization.Nuttall.getIndex(4, 8, 2.0));
+        
+        // check applied apodization
+        double[] testdata = new double[] { 1.4, 1.3, 1.1, 4.0, 2.1, 0.1, -0.2, -0.4 };
+        Apodization.apodize(testdata, Apodization.FlatTop);
+        assertArrayEquals(
+                new double[] { 0.005600000000000121, -0.2205351270430622, 0.04977851282784238, 14.489556847751528,
+                        7.607017345069555, 0.004525319347985736, 0.033928481083548026, -0.0016000000000000348 },
+                testdata);
+    }
+
+    @Test
+    public void testZeroLength() {
+        assertArrayEquals(new double[0], Apodization.Rectangular.getWindow(0));
+        assertArrayEquals(new double[0], Apodization.Blackman.getWindow(0));
+        assertArrayEquals(new double[0], Apodization.BlackmanHarris.getWindow(0));
+        assertArrayEquals(new double[0], Apodization.BlackmanNuttall.getWindow(0));
+        assertArrayEquals(new double[0], Apodization.Exponential.getWindow(0));
+        assertArrayEquals(new double[0], Apodization.FlatTop.getWindow(0));
+        assertArrayEquals(new double[0], Apodization.Hamming.getWindow(0));
+        assertArrayEquals(new double[0], Apodization.Hann.getWindow(0));
+        assertArrayEquals(new double[0], Apodization.HannExp.getWindow(0));
+        assertArrayEquals(new double[0], Apodization.Nuttall.getWindow(0));
+    }
+    
+    @Test void testCaching() {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use explicit scoping instead of the default package private level](https://app.codacy.com/manual/GSI/chart-fx/pullRequest?prid=5084368)